src/librustc_mir/transform/qualify_consts.rs

```diff
@@ -552,7 +556,11 @@ impl<'a, 'tcx> Visitor<'tcx> for Qualifier<'a, 'tcx, 'tcx> {
                     this.super_place(place, context, location);
                     match proj.elem {
                         ProjectionElem::Deref => {
-                            this.add(Qualif::NOT_CONST);
+                            if context.is_mutating_use() {
+                                this.not_const()
+                            } else {
+                                this.qualif = Qualif::NOT_CONST;
```

Comment:
Is this intentionally overriding all other `Qualif` flags? (vs invoking `add`)? Why is that? Maybe a comment?

Reply:
That is an accidental leftover from the larger refactorings. I undid the change and added comments